src/Sql/Sql/Auditing/Cmdlet/SetAzSqlDatabaseAuditing.cs

```diff
@@ -0,0 +1,426 @@
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = AuditingHelpMessages.DatabaseNameHelpMessage)]
+        [Parameter(
+            ParameterSetName = DefinitionsCommon.StorageAccountSubscriptionIdParameterSetName,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = AuditingHelpMessages.DatabaseNameHelpMessage)]
+        [Parameter(
+            ParameterSetName = DefinitionsCommon.EventHubParameterSetName,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = AuditingHelpMessages.DatabaseNameHelpMessage)]
+        [Parameter(
+            ParameterSetName = DefinitionsCommon.LogAnalyticsParameterSetName,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = AuditingHelpMessages.DatabaseNameHelpMessage)]
+        [ResourceNameCompleter("Microsoft.Sql/servers/databases", "ResourceGroupName", "ServerName")]
+        [ValidateNotNullOrEmpty]
+        public override string DatabaseName { get; set; }
+
+        [Parameter(
+            ParameterSetName = DefinitionsCommon.BlobStorageByParentResourceParameterSetName,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = AuditingHelpMessages.DatabaseInputObjectHelpMessage)]
+        [Parameter(
+            ParameterSetName = DefinitionsCommon.StorageAccountSubscriptionIdByParentResourceParameterSetName,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = AuditingHelpMessages.DatabaseInputObjectHelpMessage)]
+        [Parameter(
+            ParameterSetName = DefinitionsCommon.EventHubByParentResourceParameterSetName,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = AuditingHelpMessages.DatabaseInputObjectHelpMessage)]
+        [Parameter(
+            ParameterSetName = DefinitionsCommon.LogAnalyticsByParentResourceParameterSetName,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = AuditingHelpMessages.DatabaseInputObjectHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public override AzureSqlDatabaseModel InputObject { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = AuditingHelpMessages.StateHelpMessage)]
+        [ValidateSet(SecurityConstants.Enabled, SecurityConstants.Disabled, IgnoreCase = false)]
+        [ValidateNotNullOrEmpty]
+        public string State { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = AuditingHelpMessages.PassThruHelpMessage)]
+        public SwitchParameter PassThru { get; set; }
+
```

Comment:
The param looks [inconsistent](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/azure-powershell-design-guidelines.md#returning-no-output) with the cmdlet output type.

> In the case where your cmdlet doesn't return any output (e.g., removing, starting, stopping a resource), the cmdlet should implement the PassThru parameter and the OutputType should be set to bool

Reply:
Already tried to modify output type, static analysis failed since it is a breaking change. Lets wait for next breaking change release, and modify output type.

Reply:
You need to [deprecate output type](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-attribute-help.md#cmdletoutputbreakingchangeattribute).

Reply:
Done.